src/view/screens/Home.tsx

```diff
@@ -68,77 +48,54 @@ export function HomeScreen(props: Props) {
 
 function HomeScreenReady({
   preferences,
-  pinnedFeeds,
-  initialPage,
+  pinnedFeedInfos,
+  rawInitialFeed,
 }: Props & {
   preferences: UsePreferencesQueryResponse
-  pinnedFeeds: FeedSourceInfo[]
-  initialPage: string
+  pinnedFeedInfos: FeedSourceInfo[]
+  rawInitialFeed: string
 }) {
-  const {hasSession} = useSession()
-  const setMinimalShellMode = useSetMinimalShellMode()
-  const setDrawerSwipeDisabled = useSetDrawerSwipeDisabled()
-  const [selectedPage, setSelectedPage] = React.useState<string>(initialPage)
-
-  /**
-   * Used to ensure that we re-compute `customFeeds` AND force a re-render of
-   * the pager with the new order of feeds.
-   */
-  const pinnedFeedOrderKey = JSON.stringify(preferences.feeds.pinned)
-
-  const selectedPageIndex = React.useMemo(() => {
-    const index = ['Following', ...preferences.feeds.pinned].indexOf(
```

Comment:
It's possible this code might have been the cause of some cases where the wrong feed was being displayed. This logic searches for the selected index based on the `preferences.feeds.pinned` array. However, rendering pages was done using the `customFeeds` array (derived from `pinnedFeeds`), and their items aren't guaranteed to be in sync. So before the refactor, I think it was possible for the index to correspond to a wrong item. Which is also why clamping below was necessary.